src/main.ts

```diff
@@ -0,0 +1,242 @@
+      throw new Error(`invalid input received for gcloud_component: ${gcloudComponent}`);
+    }
+
+    // Build base command from required inputs
+    let cmd = [
+      'deploy',
+      'releases',
+      'create',
+      name,
+      '--delivery-pipeline',
+      deliveryPipeline,
+      '--region',
+      region,
+    ];
+
+    if (buildArtifacts) {
+      cmd.push('--build-artifacts', buildArtifacts);
+    }
+    if (images && Object.keys(images).length > 0) {
+      cmd.push('--images', kvToString(images));
+    }
+    if (source) {
+      cmd.push('--source', source);
+    }
+    if (disableInitialRollout) {
+      cmd.push('--disable-initial-rollout');
+    }
+    if (sourceStagingDir) {
+      cmd.push('--gcs-source-staging-dir', sourceStagingDir);
+    }
+    if (skaffoldFile) {
+      cmd.push('--skaffold-file', skaffoldFile);
+    }
+
+    const allAnnotations = Object.assign({}, getDefaultAnnotations(), annotations);
+    cmd.push('--annotations', kvToString(allAnnotations));
+
+    const allLabels = Object.assign({}, getDefaultLabels(), labels);
+    cmd.push('--labels', kvToString(allLabels));
+
+    if (description) {
+      cmd.push('--description', description);
+    }
+    if (flags) {
+      const flagList = parseFlags(flags);
+      if (flagList) {
+        cmd.push('--flags');
+        cmd = cmd.concat(flagList);
+      }
+    }
+
+    // Set output format to json for easy parsing
+    cmd.push('--format', 'json');
+
+    // Install gcloud if not already installed.
+    if (!isGcloudInstalled(gcloudVersion)) {
+      await installGcloudSDK(gcloudVersion);
+    } else {
+      const toolPath = toolCache.find('gcloud', gcloudVersion);
+      addPath(path.join(toolPath, 'bin'));
+    }
+
+    // Install gcloud component if needed and prepend the command
+    if (gcloudComponent) {
+      await installGcloudComponent(gcloudComponent);
+      cmd.unshift(gcloudComponent);
+    }
+
+    // Authenticate - this comes from google-github-actions/auth.
+    const credFile = process.env.GOOGLE_GHA_CREDS_PATH;
+    if (credFile) {
+      await authenticateGcloudSDK(credFile);
+      logInfo('Successfully authenticated');
+    } else {
+      logWarning('No authentication found, authenticate with `google-github-actions/auth`.');
+    }
+
+    const toolCommand = getToolCommand();
+    const options = { silent: true, ignoreReturnCode: true };
+    const commandString = `${toolCommand} ${cmd.join(' ')}`;
+    logInfo(`Running: ${commandString}`);
+
+    // Run gcloud cmd
+    const output = await getExecOutput(toolCommand, cmd, options);
+    if (output.exitCode !== 0) {
+      const errMsg = output.stderr || `command exited ${output.exitCode}, but stderr had no output`;
+      throw new Error(`failed to execute gcloud command \`${commandString}\`: ${errMsg}`);
+    }
+
+    const outputs: CreateCloudDeployReleaseOutputs = parseCreateReleaseResponse(output.stdout);
+
+    if (outputs.name) {
```

Comment:
When would these not be defined? It seems like it would be a bug if they weren't returned from the API, right?

Reply:
Coming via `parseCreateReleaseResponse` rather than directly from the API, I thought I'd be cautious before using these in the output.

Reply:
My only suggestion...

Should this be an error if they aren't defined instead? Or at least log a warning. The end-user might be confused as to why they have no outputs?

Reply:
Thanks both -- I now throw an error if the response from the API is empty, so have removed the check here.